cmd/otelcontribcol/go.mod

```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.22.0
 
-toolchain go1.22.7
+toolchain go1.23.1
```

Comment:
We shouldn't update the toolchain version here. (Same comment for other toolchain updates)